File: app/src/main/java/jp/juggler/fadownloader/ScanItem.java
package jp.juggler.fadownloader;

import java.util.LinkedList;

public class ScanItem{

	public final String name;
	public final String remote_path;
	public final LocalFile local_file;
	public final boolean is_file;
	public final long size;
	public final long time;
	public final String mime_type;

	public ScanItem( String name, String remote_path, LocalFile local_file ){
		this.name = name;
		this.remote_path = remote_path;
		this.local_file = local_file;
		this.is_file = false;
		this.size = 0L;
		this.time = 0L;
		this.mime_type = null;
	}

	public ScanItem( String name, String remote_path, LocalFile local_file, long size, long time, String mime_type ){
		this.name = name;
		this.remote_path = remote_path;
		this.local_file = local_file;
		this.is_file = true;
		this.size = size;
		this.time = time;
		this.mime_type = mime_type;
	}

	public static class Queue {
		public final LinkedList<ScanItem> queue_file = new LinkedList<>(  );
		public final LinkedList<ScanItem> queue_folder = new LinkedList<>(  );

		long folder_count;
		public long file_count;
		long byte_count;

		public void addFolder( ScanItem item ){
			++folder_count;
			queue_folder.add(item);
		}

		public void addFile( ScanItem item ){
			++file_count;
			byte_count += item.size;
			queue_file.add( item );
		}
	}
}

File: app/src/main/java/jp/juggler/fadownloader/model/ScanItem.kt
package jp.juggler.fadownloader.model

import jp.juggler.fadownloader.LocalFile
import java.util.LinkedList

class ScanItem(
	val name : String,
	val remote_path : String,
	val local_file : LocalFile,
	val size : Long = 0L,
	val time : Long = 0L,
	val mime_type : String
) {
	companion object {
		const val MIME_TYPE_FOLDER =  "inode/directory"
	}
	
	val is_file : Boolean = true
	
	class Queue {
		val queue_file = LinkedList<ScanItem>()
		val queue_folder = LinkedList<ScanItem>()
		var file_count : Long = 0
		
		private var folder_count : Long = 0
		private var byte_count : Long = 0
		
		fun addFolder(item : ScanItem) {
			++ folder_count
			queue_folder.add(item)
		}
		
		fun addFile(item : ScanItem) {
			++ file_count
			byte_count += item.size
			queue_file.add(item)
		}
	}
}
